joy:
  ros__parameters:
    # device_name: Sony Interactive Entertainment Wireless Controller
    # device_name: Wireless Controller
    device_id: 0
    deadzone: 0.01
    autorepeat_rate: 20.0
    coalesce_interval: 0.01

teleop_twist_joy:
  ros__parameters:
    axis_linear: # Forward/Back
      x: 1
    scale_linear:
      x: 0.5
    scale_linear_turbo:
      x: 1.0
    axis_angular: # Left/Right
      yaw: 3
    scale_angular:
      yaw: 0.5
    scale_angular_turbo:
      yaw: 0.5
    enable_button: 4
    enable_turbo_button: 6

twist_keyboard:
  ros__parameters:
    enable: false
    max_linear_speed: 2.0
    max_angular_speed: 2.0
    startup_linear_speed: 0.5
    startup_angular_speed: 1.0
    speed_control_rate: 10.0
    publish_rate: 10.0

joy_emergency_stop:
  ros__parameters:
    emergency_stop_set_joy_button_index: 5
    emergency_stop_clear_joy_button_index: 7
    joy_emergency_stop_timeout: 1.0 #seconds

twist_angular_correction:
  ros__parameters:
    topic_names: ["joy_vel"]

teleop_twist_mux:
  ros__parameters:
    topics:
      joystick:
        topic: joy_vel_corrected
        timeout: 0.5
        priority: 100
      keyboard:
        topic: key_vel
        timeout: 0.5
        priority: 90
      teleop_nav2:
        topic: teleop_nav2_vel
        timeout: 0.5
        priority: 50
